### PR TITLE
Fix reduce function

### PR DESCRIPTION
**Pull request by aholovenko**

### CODE REVIEW

**Comment by vindex10** [comment]
cool!!

1. I would test it on some VCF to make sure the output doesn't change after pickle/unpickle
2. (optional) we can make it fancier by exploiting the third parameter of __reduce__ which is responsible for appending class properties (AFAIK)

**Comment by aholovenko** [comment]
> 1. I would test it on some VCF to make sure the output doesn't change after pickle/unpickle

@vindex10 Done. Empty diff. 
I have written a script for 2 small functions: 1) just reading (vcf.Reader) and writing it (vcf.Writer), 2) reading, adding variants to queue, getting variants from queue, writing variants with writer. These were run from master and current branch. As a result I received 3 (2nd functions failed when run from master -> problem we have been solving :+1: ) identical files (nothing appeared on a diff).

File: vcf/model.py [hunk 1]
```diff
@@ -680,6 +680,14 @@ def __repr__(self):
         return str(self)
 
 
+def create_call_data(fields_dict, types, nums):
```
Comment (vindex10) [inline vcf/model.py]: please rename this. first of all to start from `_` because it is internal. also maybe more semantic:

`_restore_calldata_instance`

**Comment by vindex10** [comment]
ignore this, it is wrong:

> (optional) we can make it fancier by exploiting the third parameter of reduce which is responsible for appending class properties (AFAIK)

**Comment by amit-ezra** [comment]
@aholovenko 
can you please give me some context for this PR ?

**Comment by aholovenko** [comment]
> @aholovenko
> can you please give me some context for this PR ?

@amit-ezra 
when we get a variant from the queue  - it is not restored correctly. So I cannot process the variant as some the fields have a type of a property object and not the string.

**Review by yak1r — APPROVED** [review]

**Pull request merged**